File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.py
# def solution(numLog):
#     answer =''
#     start= 0
#     dic = {'1':'w', '-1' :'s', '10':'d', '-10':'a'}
#     for idx in range(1, len(numLog)):
#             answer+= dic[str(numLog[idx] - numLog[idx-1])]
#     return answer

def solution(numLog):
    answer =''
    start= numLog[0]
    dic = {'1':'w', '-1' :'s', '10':'d', '-10':'a'}
    for n in numLog:
        if n - start!=0:
            answer+=dic[str(n - start)]
        start = n 
    return answer

File: 프로그래머스/0/181925. 수 조작하기 2/수 조작하기 2.cpp
#include <string>
#include <vector>

using namespace std;

string solution(vector<int> numLog){
    string str="";
    int cut = 0;
    int diff;
    while(1){
        diff = numLog[cut+1] - numLog[cut];
        switch(diff){
            case 1: str += 'w'; break;
            case -1: str += 's'; break;
            case 10: str += 'd'; break;
            case -10: str += 'a';  break;
            default: break;
        }
        if(cut++>(numLog.size()-1)) break;
    }

    return str;
}